data/campaigns/fri04.wmf/scripting/mission_thread.lua

```diff
@@ -0,0 +1,173 @@
+local seen_reebaud = false
+local seen_amazons = false
+
+local o_defeat_amz = nil
+local o_defeat_emp = nil
+local amazons_defeated_by_reebaud = false
+
+function hurry_up()
+   while true do
+      sleep(10000)
+      if seen_reebaud or p4.defeated then return end
+      if #p2:get_buildings("frisians_headquarters") < 1 then
+         campaign_message_box(reebaud_in_danger)
+         return
+      end
+   end
+end
+
+function warn_empire_expansion()
+   while (
+         #p3:get_buildings("empire_blockhouse") +
+         #p3:get_buildings("empire_sentry") +
+         #p3:get_buildings("empire_outpost") +
+         #p3:get_buildings("empire_barrier") +
+         #p3:get_buildings("empire_tower") +
+         #p3:get_buildings("empire_fortress") +
+         #p3:get_buildings("empire_castle")) < 4 do
+      sleep(100000)
+      if seen_reebaud then return end
+   end
+   campaign_message_box(legate_expands)
+end
+
+function see_amazons(field)
+   if amazons_defeated_by_reebaud then return end
+   scroll_to_field(field)
+   sleep(2000)
+   if not seen_reebaud then
+      campaign_message_box(amz_1a)
+      campaign_message_box(amz_1b)
+   else
+      campaign_message_box(amz_1c)
+   end
+   sleep(500)
+   campaign_message_box(amz_2)
+   sleep(500)
+   campaign_message_box(amz_3)
+   if not o_defeat_amz then o_defeat_amz = add_campaign_objective(obj_defeat_amz) end
+   while not p4.defeated do sleep(1000) end
+   if seen_reebaud then campaign_message_box(victory_amz_a) else campaign_message_box(victory_amz_b) end
+   set_objective_done(o_defeat_amz)
+end
+
+function see_reebaud(field)
+   local empire_already_defeated = (p3.defeated or not campaign_data.payment)
+   local o_recruit_train = nil
+   scroll_to_field(field)
+   sleep(2000)
+   campaign_message_box(reebaud_1)
+   p2.team = 1
+   campaign_message_box(reebaud_2)
+   campaign_message_box(reebaud_3)
+   campaign_message_box(reebaud_4)
+   sleep(2000)
+   campaign_message_box(reebaud_5)
+   campaign_message_box(reebaud_6)
+   campaign_message_box(reebaud_7)
+   if p4.defeated then
+      amazons_defeated_by_reebaud = true
+      campaign_message_box(reebaud_8b)
+      if empire_already_defeated then
+         campaign_message_box(reebaud_9c)
+         o_recruit_train = add_campaign_objective(obj_recruit_train)
+      end
+   else
+      campaign_message_box(reebaud_8a)
+      if seen_amazons then campaign_message_box(reebaud_9a) else campaign_message_box(reebaud_9b) end
+   end
+   p3.team = 0
+   if not o_defeat_amz and not amazons_defeated_by_reebaud then o_defeat_amz = add_campaign_objective(obj_defeat_amz) end
+   if not empire_already_defeated then
+      sleep(2000)
+      campaign_message_box(reebaud_10)
+      campaign_message_box(reebaud_11)
+      campaign_message_box(reebaud_12)
+      o_defeat_emp = add_campaign_objective(obj_defeat_emp)
+      p1:set_attack_forbidden(2, false)
```

Comment:
the lines about p1 attacking p2 and vice versa are superfluous as they are in the same team